TechtonicaModLoader/ViewModels/MainViewModel.cs

```diff
@@ -102,8 +101,17 @@ private void CheckForModUpdates() {
 
         [RelayCommand]
         private void CreateNewProfile() {
+            // See note on _dialogService.   
             _dialogService.ShowInfoMessage("Test", "This is a test message", "Close Test Box");
             if(_dialogService.GetStringFromUser(out string name, "Enter Profile Name:", "")) {
+                if (_profileManager.ProfilesList.Select(profile => profile.Name).Contains(name)) {
+
+                    // TODO: Decision to make:
+                    // Create an message for the View to pick up and show the confirmation UX.
```

Comment:
After thinking about approaches, I should have said: Have a View - ViewModel interaction to handle this case.
The ideal here would be to have ViewModel properties that could be set and examined.   I think the messaging approach can work but gets messy.

Reply:
Do you see why I'm worrying about this code from a MVVM perspective and what I mean by 'a View - ViewModel interaction'?